packages/yoroi-extension/app/utils/recoveryPhrase.js

```diff
@@ -0,0 +1,29 @@
+// @flow
+
+export type SortedRecoveryPhraseEntry = {|
+    id: number,
+    word: string,
+    originalIdx: number,
+|};
+
+export function makeSortedPhrase(recoveryPhrase: Array<string>): SortedRecoveryPhraseEntry[] {
+  const sorted = recoveryPhrase.slice().sort();
+  const wordIndexes = new Set();
+
+  return sorted.map((sortedWord, sortedWordIdx) => {
+    const originalIdx = recoveryPhrase.findIndex((originalWord, idx) => {
+      return sortedWord === originalWord && !wordIndexes.has(idx)
+    });
+
+    if (originalIdx === -1) throw new Error('Word not found in the original recovery phrase. Should never happen');
+
+    // Mark word index as watched to handle recovery phrase with duplicates
+    wordIndexes.add(originalIdx);
```

Comment:
@AhmedIbrahim336 , you can avoid having to check for duplicates separately if you just store original indexes before sorting and then sort by a specific condition, might make it a bit more readable, e.g.:

```
const sorted = recoveryPhrase
  .map((word, originalIndex) => ({ word, originalIndex }))
  .sort((a, b) => a.word.localeCompare(b.word))
  .map((word, id) => ({ ...word, id }));
```